rules/prefer-event-target.js

```diff
@@ -6,26 +6,16 @@ const messages = {
 	[MESSAGE_ID]: 'Prefer `EventTarget` over `EventEmitter`.',
 };
 
-const eventEmitterSuperClassSelector = [
-	matches(['ClassDeclaration', 'ClassExpression']),
-	'[body.type="ClassBody"]',
-	' > ',
-	'Identifier.superClass',
+const selector = [
+	'Identifier',
 	'[name="EventEmitter"]',
+	matches([
+		'ClassDeclaration > .superClass',
+		'ClassExpression > .superClass',
+		'NewExpression > .callee',
+	]),
```

Comment:
I don't think this rule cases about `ClassBody`, so I just removed it. @jopemachine